### DISCUSSION TITLE
DataTable multiple rows selected, while providing one row id

### DISCUSSION THREAD

**Discussion by bhavers** [carbon-design-system/carbon-components-svelte#1808] [Q&A] [answered]
Hi,
My DataTable shows multiple rows selected, while i just expect one to be selected.

I am providing a store `$selectedId` to a DataTable's `selectedRowIds` attribute. The expected behaviour is that the DataTable selects the row that is specified in `$selectedId`. For most rows this works, but sometimes it selects multiple rows (the class `bx--data-table--selected` is added to the row).
In the provided example below, selecting "ABC" would also select rows "Personal" and "Report". Selecting "Read" will also select "Report". The others rows acts as expected when selected (only the selected row is selected).
When you change the id of the item in the supplied data, it changes the outcome. For example, if you change in the `testData.js` the `id` of Personal from 3 to 33, than it will not be selected anymore when selecting ABC.
I must be missing something simple here, but can't figure it out. Any help is appreciated. Thx.

+page.svelte
```
<script>
  import { DataTable } from 'carbon-components-svelte';
  import { selectedId, data } from './store.js';
  import 'carbon-components-svelte/css/all.css';
</script>

<DataTable
  stickyHeader
  selectedRowIds={$selectedId}
  size="short"
  headers={[
    { key: 'description', value: 'Description' },
    { key: 'parentId', value: 'ParentId'},
  ]}
  rows={$data}
  on:click:row={({ detail }) => {
    $selectedId = detail.id;
  }}
>
</DataTable>
```

store.js
```
import { writable } from 'svelte/store';
import { testData } from './testData';

export const selectedId = writable(); 
export const data = writable(testData);
```

testData.js
```
export const testData = 
[
  {
    id: '7',
    description: 'Sub sub',
    parentId: '3',
  },
  {
    id: '8',
    description: 'Sub sub Sub',
    parentId: '7',
  },
  {
    id: '1',
    description: 'RootA',
    created: '2023-05-30 21:00',
    parentId: '',
  },
  {
    id: '47',
    description: 'Read',
    parentId: '43',
  },
  {
    id: '43',
    description: 'ABC',
    parentId: '2',
  },
  {
    id: '2',
    description: 'Work',
    parentId: '1',
  },
  {
    id: '3',
    description: 'Personal',
    parentId: '1',
  },
  {
    id: '4',
    description: 'Report',
    parentId: '2',
  },
  {
    id: '55',
    description: 'Close X',
    parentId: '2',
  },
  {
    id: '6',
    description: 'Clean Y',
    parentId: '3',
  },
  {
    id: '9',
    description: 'RootB',
    parentId: '',
  }
];```

**Comment by metonym** [comment, marked as answer]
The `selectedRowIds` expects an array; can you try the following?

```diff
- selectedRowIds={$selectedId}
+ selectedRowIds={[$selectedId]}
```

**Comment by bhavers** [comment]
omg! i have been staring at this for two evenings, so simple.
Thanks a lot.
